tests/unit/test_network_connections.py

```diff
@@ -2786,13 +2792,176 @@
+                    "zone": "external",
+                    "mac": "12:23:34:45:56:60",
+                    "ip": {
+                        "dns_search": ["aa", "bb"],
+                        "route_append_only": True,
+                        "rule_append_only": False,
+                        "route": [
+                            {"network": "192.168.45.0", "metric": 545},
+                            {"network": "192.168.46.0", "prefix": 30},
+                            {"network": "a:b:c:d::"},
+                        ],
+                    },
+                }
+            ],
+            nm_route_list_current=[
+                [
+                    {"network": "192.168.40.0", "prefix": 24, "metric": 545},
+                    {"network": "192.168.46.0", "prefix": 30},
+                    {"network": "a:b:c:f::"},
+                ]
+            ],
+            nm_route_list_expected=[
+                [
+                    {"network": "192.168.40.0", "prefix": 24, "metric": 545},
+                    {"network": "192.168.46.0", "prefix": 30},
+                    {"network": "192.168.45.0", "prefix": 24, "metric": 545},
+                    {"network": "a:b:c:f::"},
+                    {"network": "a:b:c:d::"},
+                ]
+            ],
+            initscripts_content_current=[
+                {
+                    "ifcfg": "",
+                    "keys": None,
+                    "route": "192.168.40.0/24 metric 545\n192.168.46.0/30",
+                    "route6": "a:b:c:f::/64",
+                    "rule": None,
+                    "rule6": None,
+                }
+            ],
+            initscripts_dict_expected=[
+                {
+                    "ifcfg": {
+                        "BOOTPROTO": "dhcp",
+                        "DOMAIN": "aa bb",
+                        "HWADDR": "12:23:34:45:56:60",
+                        "IPV6INIT": "yes",
+                        "IPV6_AUTOCONF": "yes",
+                        "NM_CONTROLLED": "no",
+                        "ONBOOT": "yes",
+                        "TYPE": "Ethernet",
+                        "ZONE": "external",
+                    },
+                    "keys": None,
+                    "route": "192.168.40.0/24 metric 545\n"
+                    "192.168.46.0/30\n"
                     "192.168.45.0/24 metric 545\n",
                     "route6": "a:b:c:f::/64\na:b:c:d::/64\n",
                     "rule": None,
                     "rule6": None,
                 }
             ],
+            initscripts_expected_warnings=[
+                "The connection e556 does not specify an interface name. Therefore, the route "
+                "to 192.168.45.0/24 will be configured without the output device and the "
+                "kernel will choose it automatically which might result in an unwanted device "
+                "being used. To avoid this, specify `interface_name` in the connection "
+                "appropriately.",
```

Comment:
It seems worth to create a template string for the warning and then format it accordingly for each different route to reduce the duplication. It will also solve the CodeQL warning.

Reply:
I created the template string to reduce the duplication.